app/(tabs)/profile.tsx

```diff
@@ -0,0 +1,81 @@
+import { router, useFocusEffect } from 'expo-router';
+import React, { useCallback } from 'react';
+import { Alert, Pressable, StyleSheet, Text, View } from 'react-native';
+import type { LogoutResponseBodyGet } from '../(auth)/api/logout+api';
+import { colors } from '../../constants/colors';
+import type { UserResponseBodyGet } from '../api/user+api';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: colors.background,
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  button: {
+    marginTop: 30,
+    backgroundColor: colors.text,
+    padding: 12,
+    borderRadius: 12,
+    shadowColor: colors.white,
+    shadowOffset: { width: 0, height: 2 },
+    shadowOpacity: 0.2,
+    shadowRadius: 4,
+    elevation: 4,
+    width: '100%',
+  },
+  text: {
+    fontFamily: 'Poppins_400Regular',
+    color: colors.cardBackground,
+    textAlign: 'center',
+    fontSize: 18,
+  },
+});
+
+export default function Profile() {
+  useFocusEffect(
+    useCallback(() => {
+      async function getUser() {
+        const response = await fetch('/api/user');
+
+        const body: UserResponseBodyGet = await response.json();
+
+        if ('error' in body) {
+          Alert.alert('Error', body.error, [{ text: 'OK' }]);
+          return router.push('/(auth)/login');
+        }
+      }
+      getUser().catch((error) => {
+        console.error(error);
+      });
+    }, []),
+  );
+
+  return (
+    <View style={styles.container}>
+      <Pressable
+        style={({ pressed }) => [styles.button, { opacity: pressed ? 0.5 : 1 }]}
+        onPress={async () => {
+          const response = await fetch('/api/logout');
+
+          if (!response.ok) {
+            let errorMessage = 'Error logging out';
+            try {
+              const responseBody: LogoutResponseBodyGet = await response.json();
+              if ('error' in responseBody) {
+                errorMessage = responseBody.error;
+              }
+            } catch {}
```

Comment:
Let's get rid of the error-swallowing `try/catch` here

Reply:
removed in this commit [40d30a4](https://github.com/upleveled/expo-example-fall-2024-atvie/pull/1/commits/40d30a420347fe0347f873cc693ff8f7d5ed3e9a)

From register, removed here [778a846](https://github.com/upleveled/expo-example-fall-2024-atvie/pull/1/commits/778a84620d263bbe7b7eb3ca30b4d196b3d77cd9)